app/controllers/users_controller.rb

```diff
@@ -4,17 +4,24 @@ class UsersController < ActionController::Base
         :invalid_email => "invalid email",
         :invalid_password => "invalid password",
         :email_exists => "email already exists",
+        :user_not_found => "user not found"
     }
 
     def authenticate
         token = params[:token]
 
-        token_results = User.parse_access_token(token)
+        begin
+            token_results = User.parse_access_token(token)
+        rescue
+            render :json => { :error => :failed }
```

Comment:
Maybe add an entry to `ERROR_MESSAGES` so that this becomes `render :json => { :error => ERROR_MESSAGES[:invalid_token] }` or something.